src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1631,13 +1631,18 @@ private int getEntryPos(String name, boolean addSlash) {
                         // slash
                         int entryLen = entry.length();
                         int nameLen = name.length();
-                        if ((entryLen == nameLen && entry.equals(name)) ||
-                                (addSlash &&
-                                nameLen + 1 == entryLen &&
-                                entry.startsWith(name) &&
-                                entry.charAt(entryLen - 1) == '/')) {
+                        if (entryLen == nameLen && entry.equals(name)) {
+                            // Found our match
                             return pos;
                         }
+                        // If addSlash is true we'll now test for name+/ providing
+                        if (addSlash && nameLen + 1 == entryLen
+                                && entry.startsWith(name) &&
+                                entry.charAt(entryLen - 1) == '/') {
+                            // Now check for a match with a trailing slash
```

Comment:
Hello Lance,
Is this a typo in that comment? Should it instead say "... without a trailing slash"?

Reply:
Hi Jaikiran,

The comment is correct

Reply:
Thank you for that clarification.  The "addSlash" param being "false" in the call below that comment is what made me think that the comment had a typo. I read that code in a bit more detail now and I see what that comment means. It essentially says that it's now issuing a call to find a entry position (if it exists) for the name ending with a "/" character.

Reply:
The comment is a bit too fuzzy. What's happening here is we searched for 'foo' and found a match for `foo/` - but we can't be sure there's no `foo` entry later on in the hash bucket, so we (somewhat inefficiently) redo the search without the trailing slash.

I suggest clarifying to something like this:
```  // "name+/" entry found, keep looking for and prefer an exact match```

Reply:
I just tweaked the comment slightly